packages/s3-browser/README.md

```diff
@@ -60,33 +171,107 @@ function MyS3Browser() {
 }
 ```
 
-### S3 Utility Functions
+### S3CredentialForm Component
 
-The package also provides utility functions for working with S3:
+The `S3CredentialForm` component provides a form interface for managing S3 credentials and saved connections.
 
 ```tsx
-import {S3Client} from '@aws-sdk/client-s3';
-import {
-  listFilesAndDirectoriesWithPrefix,
-  deleteS3Files,
-} from '@sqlrooms/s3-browser';
-
-// Initialize S3 client
-const s3Client = new S3Client({region: 'us-east-1'});
-
-// List files and directories
-async function listFiles() {
-  const files = await listFilesAndDirectoriesWithPrefix(
-    s3Client,
-    'my-bucket',
-    'path/to/directory',
+import {S3CredentialForm} from '@sqlrooms/s3-browser';
+import {S3Config, S3Connection} from '@sqlrooms/s3';
+
+function MyS3ConnectionManager() {
+  const handleConnect = async (credentials: S3Config) => {
```

Comment:
Same here, I think all this logic could be encapsulated within the slice.  For loading/saving I think it would be better to just add `initialCredentials` and `onSaveCredentials` props to `createS3BrowserSlice`.